.github/workflows/ci.yml

```diff
@@ -28,18 +28,6 @@ jobs:
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
-      - name: Gems cache
-        uses: actions/cache@v2
-        with:
-          path: ~/gems
-          key: gems-${{ matrix.ruby }}-${{ hashFiles('*.gemspec', 'Gemfile') }}-${{ github.sha }}
-          restore-keys: |
-            gems-${{ matrix.ruby }}-${{ hashFiles('*.gemspec', 'Gemfile') }}-
-            gems-${{ matrix.ruby }}-
-      - name: Install dependencies
-        run: |
-          gem install bundler
-          bundle config path ~/gems
-          bundle install --jobs 4 --retry 3
+          bundler-cache: true
```

Comment:
I admit that I didn't know about this option. However, from a cursory look it might not suite well Prawn. I don't want source tree to be modified. Bundling into the source root is explicitly not that.

For one, it confuses Rubocop. I suspect it might confuse other tools as well and I'd rather not deal with that.

Readme also claims:
> it is verbose and _very difficult_ to use a correct cache key

I agree, but I'm not convinced the action does the right thing for the gem case. For example, we don't have `Gemfile.lock` in the repo and our `Gemfile` is essentially never changes because it's just a reference to the gemspec. Will the action pick an appropriate cache key taking into account that?

Reply:
> I agree, but I'm not convinced the action does the right thing for the gem case. For example, we don't have `Gemfile.lock` in the repo and our `Gemfile` is essentially never changes because it's just a reference to the gemspec.

Yes, it will (I know, I'm the maintainer of it). It works by generated a Gemfile.lock first, just like `bundle install` would.
And then the key is based on the hash of that generated Gemfile.lock, and everything else (many things) relevant for caching.

> Will the action pick an appropriate cache key taking into account that?

Yes, and better than that it will also automatically account for e.g. changing ABI for ruby-head (yet the ABI version doesn't change), etc.
Maybe you got confused by https://bibwild.wordpress.com/2020/11/12/deep-dive-moving-ruby-projects-from-travis-to-github-actions-for-ci/ ? That's unfortunately outdated by now, all these issue have been long fixed.

The previous key used in this workflow has several problems, ignoring the OS version (could be a problem when `ubuntu-latest` points to something newer), restores keys without `bundle clean` which could grow the cache over time, doesn't properly hash based on the Gemfile.lock, etc.

I wonder why sometimes people think they know better than the ruby org about Bundler caching. (not directed at you, just an observation)